test/ng/directive/asSpec.js

```diff
@@ -0,0 +1,87 @@
+'use strict';
+
+describe('as', function() {
+
+  it('should bind in the current scope the controller of a component', function() {
+    var expected = {some: 'value'};
+
+    angular.module('my', []).component('myComponent', {
+      template: 'foo',
+      controller: function() {
+        this.property = expected;
+      }
+    });
+    module('my');
+
+    inject(function($compile, $rootScope) {
+      var scope = $rootScope.$new();
+      var $ctrl = {undamaged: true};
+      scope.$ctrl = $ctrl;
+      dealoc($compile('<my-component as="$ctrl.myComponent"></my-component>')(scope));
+
+      expect(scope.$ctrl).toBe($ctrl);
+      expect(scope.$ctrl.undamaged).toBe(true);
+      expect($ctrl.myComponent).not.toBeUndefined();
+      expect($ctrl.myComponent && $ctrl.myComponent.property).toBe(expected);
+      scope.$destroy();
+    });
+  });
+
+  it('should be parametrized with any variable', function() {
+    var myComponentController;
+
+    angular.module('my', []).component('myComponent', {
+      template: 'foo',
+      controller: function() {
+        myComponentController = this;
+      }
+    });
+    module('my');
+
+    inject(function($compile, $rootScope) {
+      var scope = $rootScope.$new();
+      dealoc($compile('<my-component as="bar.myComponent"></my-component>')(scope));
+
+      expect(scope.bar).not.toBeUndefined();
+      expect(scope.bar && scope.bar.myComponent).toBe(myComponentController);
+      scope.$destroy();
+    });
+  });
+
+  it('should be compatible with entity directives with controller', function() {
+    var myDirectiveController;
+
+    angular.module('my', []).directive('myDirective', function() {
+      return {
+        restrict: 'E',
+        controller: function() {
+          myDirectiveController = this;
+        }
+      };
+    });
+    module('my');
+
+    inject(function($compile, $rootScope) {
+      var scope = $rootScope.$new();
+      dealoc($compile('<my-directive as="bar.myDirective"></my-directive>')(scope));
+
+      expect(scope.bar).not.toBeUndefined();
+      expect(scope.bar && scope.bar.myDirective).toBe(myDirectiveController);
+      scope.$destroy();
+    });
+  });
+
+  it('should bind non HTMLElements elements into the current scope if no component', inject(function($compile, $rootScope) {
```

Comment:
It has a problem. It is modifying the module definition in each test
(creating a special directive case in each test). So it can inject nothing
before module definition. I copied this solution from .component testing.
(I do not put the example because I'm on mobile now)

El dv, 19 febr 2016 a les 13:55 Georgios Kalpakas <notifications@github.com>
va escriure:

> In test/ng/directive/asSpec.js
> <https://github.com/angular/angular.js/pull/14080#discussion_r53456808>:
>
> > +        }
> > +      };
> > +    });
> > +    module('my');
> > +
> > +    inject(function($compile, $rootScope) {
> > +      var scope = $rootScope.$new();
> > +      dealoc($compile('<my-directive as="bar.myDirective"></my-directive>')(scope));
> > +
> > +      expect(scope.bar).not.toBeUndefined();
> > +      expect(scope.bar && scope.bar.myDirective).toBe(myDirectiveController);
> > +      scope.$destroy();
> > +    });
> > +  });
> > +
> > +  it('should bind non HTMLElements elements into the current scope if no component', inject(function($compile, $rootScope) {
>
> Intead of injecting $rootScope and $compile every time, it's cleaner to
> do it in a beforeEach() block.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/angular/angular.js/pull/14080/files#r53456808>.
>